tuning/sft_trainer.py

```diff
@@ -238,12 +250,34 @@ def train(
     }
 
     json_dataset = datasets.load_dataset("json", data_files=data_files)
-    formatted_train_dataset = json_dataset["train"].map(format_dataset)
+    if data_args.data_formatter_template:
+        (
+            formatted_train_dataset,
+            data_args.dataset_text_field,
+        ) = apply_custom_formatting_template(
+            json_dataset["train"],
+            data_args.data_formatter_template,
+            tokenizer.eos_token,
+        )
+    else:
+        formatted_train_dataset = json_dataset["train"].map(format_dataset)
     logger.info("Training dataset length is %s", len(formatted_train_dataset))
 
     formatted_validation_dataset = None
     if data_args.validation_data_path:
-        formatted_validation_dataset = json_dataset["validation"].map(format_dataset)
+        if data_args.data_formatter_template:
```

Comment:
there will need to be some consolidation with my PR, just FYI - yours will be merged first, but the calling of this is probably going to get shuffled into some utils to make this a bit easier to manage with different collators